### DISCUSSION TITLE
Text Selection Feature in Commentaries

### DISCUSSION THREAD

**Discussion by mrnazu** [ezra-bible-app/ezra-bible-app#1069] [Ideas]
Hey, based on my experience using this blessed based app, I’d like to propose a new feature that allows users to select and copy text within the commentaries. Currently, the interface restricts text selection, which can be inconvenient for users who want to quote or reference specific passages or even some Hebrew words. (I know in the new version which is v1.14 , released 3 days ago, you guys included the copy button, but that's not enough, right?)

**Feature Details:**
- Enable text selection within commentary sections.

I mean, selecting specific words, sentences, paragraphs, or just text as we normally do in social media, a browser, or any other application. Then, we should be able to copy that text without needing a dedicated button. I know improving the `user-select: none;` property might solve this. to `user-select: text;` or simply removing the `user-select`. Should I contribute to this improvement?

**Comment by tobias-klein** [comment]
Hi @mrnazu!

Thanks once more for your initiative. Something we may do on top of the new functionality is to react to the keyboard shortcut ctrl+c or command+c when the commentary text is selected.

The code that handles this shortcut is currently implemented in `app_controller.js` and looks like this:

```JavaScript

  initGlobalShortCuts() {
    let shortCut = 'ctrl+c';
    if (platformHelper.isMac()) {
      shortCut = 'command+c';
    }

    Mousetrap.bind(shortCut, async () => {
      await this.verse_selection.copySelectedVerseTextToClipboard();
      return false;
    });
```

What we would have to do is to check whether the dictionary selection is active and if that is the case, call the function for copying the dictionary text to the clipboard - instead of calling the function visible above. Let me know if you are interested to supply another PR for that change. From the context of `app_controller` the commentary panel API is reachable via `app_controller.commentaryPanel.<function-name>`.

**Reply by mrnazu** [reply]
Okay sure. I’ll create a new pull request.

**Comment by mrnazu** [comment]
I have already updated the `initGlobalShortCuts` method to include this functionality:

- **Check if Dictionary Selection is Active**:
  ```javascript
  if (this.dictionary_controller.isSelectionActive()) {
    await this.dictionary_controller.copySelectedTextToClipboard();
  }
  ```
  - **Task**: If the dictionary selection is active, copy the selected dictionary text to the clipboard.

- **Else Condition for Verse Selection**:
  ```javascript
  else {
    await this.verse_selection.copySelectedVerseTextToClipboard();
  }
  ```
  - **Task**: If the dictionary selection is not active, copy the selected verse text to the clipboard.

This logic is added within the existing `Mousetrap.bind(shortCut, async () => { ... })` block.

With these changes, the application will now handle the `ctrl+c` or `command+c` shortcut appropriately depending on whether the dictionary selection is active. If you have any further suggestions or need adjustments, feel free to let me know.

**Comment by mrnazu** [comment]
Here is a short summary of the changes made to the `AppController` class:

- **New Properties in Constructor**:
  ```javascript
  this.dictionarySelectionActive = false;
  ```
  - **Task**: Initialize `dictionarySelectionActive` property to track if dictionary selection is active.

- **New Method: isDictionarySelectionActive**:
  ```javascript
  isDictionarySelectionActive() {
    return this.dictionarySelectionActive;
  }
  ```
  - **Task**: Check if dictionary selection is active.

- **New Method: setDictionarySelectionActive**:
  ```javascript
  setDictionarySelectionActive(active) {
    this.dictionarySelectionActive = active;
  }
  ```
  - **Task**: Set the state of dictionary selection to active or inactive.
